--- a/examodels.jl	2026-09-27 23:35:47
+++ b/examodels.jl	2026-09-27 23:35:47
@@ -105,40 +105,43 @@
 
     w = ExaModels.ExaCore()
 
-    va = ExaModels.variable(w, length(data.bus);)
+    ExaModels.@add_var(w, va, length(data.bus);)
 
-    vm = ExaModels.variable(
+    ExaModels.@add_var(
         w,
+        vm,
         length(data.bus);
         start = fill!(similar(data.bus, Float64), 1.0),
         lvar = data.vmin,
         uvar = data.vmax,
     )
-    pg = ExaModels.variable(w, length(data.gen); lvar = data.pmin, uvar = data.pmax)
+    ExaModels.@add_var(w, pg, length(data.gen); lvar = data.pmin, uvar = data.pmax)
 
-    qg = ExaModels.variable(w, length(data.gen); lvar = data.qmin, uvar = data.qmax)
+    ExaModels.@add_var(w, qg, length(data.gen); lvar = data.qmin, uvar = data.qmax)
 
-    p = ExaModels.variable(w, length(data.arc); lvar = -data.rate_a, uvar = data.rate_a)
+    ExaModels.@add_var(w, p, length(data.arc); lvar = -data.rate_a, uvar = data.rate_a)
 
-    q = ExaModels.variable(w, length(data.arc); lvar = -data.rate_a, uvar = data.rate_a)
+    ExaModels.@add_var(w, q, length(data.arc); lvar = -data.rate_a, uvar = data.rate_a)
 
-    o = ExaModels.objective(
+    o = ExaModels.@add_obj(
         w,
         g.cost1 * pg[g.i]^2 + g.cost2 * pg[g.i] + g.cost3 for g in data.gen
     )
 
-    c1 = ExaModels.constraint(w, va[i] for i in data.ref_buses)
+    ExaModels.@add_con(w, c1, va[i] for i in data.ref_buses)
 
-    c2 = ExaModels.constraint(
+    ExaModels.@add_con(
         w,
+        c2,
         p[b.f_idx] - b.c5 * vm[b.f_bus]^2 -
         b.c3 * (vm[b.f_bus] * vm[b.t_bus] * cos(va[b.f_bus] - va[b.t_bus])) -
         b.c4 * (vm[b.f_bus] * vm[b.t_bus] * sin(va[b.f_bus] - va[b.t_bus])) for
         b in data.branch
     )
 
-    c3 = ExaModels.constraint(
+    ExaModels.@add_con(
         w,
+        c3,
         q[b.f_idx] +
         b.c6 * vm[b.f_bus]^2 +
         b.c4 * (vm[b.f_bus] * vm[b.t_bus] * cos(va[b.f_bus] - va[b.t_bus])) -
@@ -146,16 +149,18 @@
         b in data.branch
     )
 
-    c4 = ExaModels.constraint(
+    ExaModels.@add_con(
         w,
+        c4,
         p[b.t_idx] - b.c7 * vm[b.t_bus]^2 -
         b.c1 * (vm[b.t_bus] * vm[b.f_bus] * cos(va[b.t_bus] - va[b.f_bus])) -
         b.c2 * (vm[b.t_bus] * vm[b.f_bus] * sin(va[b.t_bus] - va[b.f_bus])) for
         b in data.branch
     )
 
-    c5 = ExaModels.constraint(
+    ExaModels.@add_con(
         w,
+        c5,
         q[b.t_idx] +
         b.c8 * vm[b.t_bus]^2 +
         b.c2 * (vm[b.t_bus] * vm[b.f_bus] * cos(va[b.t_bus] - va[b.f_bus])) -
@@ -163,32 +168,35 @@
         b in data.branch
     )
 
-    c6 = ExaModels.constraint(
+    ExaModels.@add_con(
         w,
+        c6,
         va[b.f_bus] - va[b.t_bus] for b in data.branch;
         lcon = data.angmin,
         ucon = data.angmax,
     )
-    c7 = ExaModels.constraint(
+    ExaModels.@add_con(
         w,
+        c7,
         p[b.f_idx]^2 + q[b.f_idx]^2 - b.rate_a_sq for b in data.branch;
         lcon = fill!(similar(data.branch, Float64, length(data.branch)), -Inf),
     )
-    c8 = ExaModels.constraint(
+    ExaModels.@add_con(
         w,
+        c8,
         p[b.t_idx]^2 + q[b.t_idx]^2 - b.rate_a_sq for b in data.branch;
         lcon = fill!(similar(data.branch, Float64, length(data.branch)), -Inf),
     )
 
-    c9 = ExaModels.constraint(w, b.pd + b.gs * vm[b.i]^2 for b in data.bus)
+    ExaModels.@add_con(w, c9, b.pd + b.gs * vm[b.i]^2 for b in data.bus)
 
-    c10 = ExaModels.constraint(w, b.qd - b.bs * vm[b.i]^2 for b in data.bus)
+    ExaModels.@add_con(w, c10, b.qd - b.bs * vm[b.i]^2 for b in data.bus)
 
-    c11 = ExaModels.constraint!(w, c9, a.bus => p[a.i] for a in data.arc)
-    c12 = ExaModels.constraint!(w, c10, a.bus => q[a.i] for a in data.arc)
+    c11 = ExaModels.@add_con!(w, c9, a.bus => p[a.i] for a in data.arc)
+    c12 = ExaModels.@add_con!(w, c10, a.bus => q[a.i] for a in data.arc)
 
-    c13 = ExaModels.constraint!(w, c9, g.bus => -pg[g.i] for g in data.gen)
-    c14 = ExaModels.constraint!(w, c10, g.bus => -qg[g.i] for g in data.gen)
+    c13 = ExaModels.@add_con!(w, c9, g.bus => -pg[g.i] for g in data.gen)
+    c14 = ExaModels.@add_con!(w, c10, g.bus => -qg[g.i] for g in data.gen)
 
     model = ExaModels.TimedNLPModel(
         ExaModels.ExaModel(w)
